cli/flox/src/commands/push.rs

```diff
@@ -206,3 +212,18 @@ impl Push {
         "}
     }
 }
+
+/// Ensure that the environment does not include local includes before pushing it to FloxHub
+fn check_for_local_includes(flox: &Flox, env: &impl Environment) -> Result<(), anyhow::Error> {
+    let manifest = env.manifest(flox)?;
+    let has_local_include = manifest
+        .include
+        .environments
+        .iter()
+        .any(|include| matches!(include, IncludeDescriptor::Local { .. }));
+
+    if has_local_include {
+        bail!("cannot push environment that includes local environments.")
+    }
+    Ok(())
+}
```

Comment:
I debated whether this should be a cli restriction or denied on a lower level by the environment op(s).

I ended up choosing the cli side for expedience and to stick closer to the mentioned output format, but happy to push this further down if needed.

Reply:
Good call out. There's a chance that we'll forget to apply the restriction somewhere else that it matters. Is the fiddly part threading new errors through?

Reply:
well, i think this wouldnt be that fiddly in the implemetation, bubbling up the error is relatively straight forward, it would just either look differently (as there is a bit of a chain of errors in that case) or needs special casing to extract that specific error.
I'd personally lean more towards bubbling it up having thought about it a bit.

Reply:
moved this to the manaifest/push implemetation